web/modules/custom/joinup_notification/src/JoinupMessageDeliveryInterface.php

```diff
@@ -5,145 +5,142 @@
+   * If the message entity is not saved, the service will take care to save it
+   * prior to delivery.
    *
-   * @param string $message_template
-   *   The message template ID.
-   * @param array $values
-   *   (optional) Additional values to be passed to the message entity creation,
-   *   similar to values passed to EntityInterface::create().
+   * This is intended to be used only when sending messages to recipients which
+   * are not registered users of the website. If the recipient is a registered
+   * user, use ::sendMessageToMultipleUsers() instead.
    *
-   * @return $this
+   * @param \Drupal\message\MessageInterface $message
+   *   The message to be delivered.
+   * @param string[] $mails
+   *   A list of e-mail addresses to send the message to.
+   * @param array $notifier_options
+   *   An optional associative array of options to pass to the Email notifier
+   *   plugin.
    *
-   * @see \Drupal\Core\Entity\EntityInterface::create()
-   * @see self::setMessage()
+   * @return bool
+   *   Whether or not the messages were sent successfully.
    */
-  public function createMessage(string $message_template, array $values = []): self;
+  public function sendMessageToEmailAddresses(MessageInterface $message, array $mails, array $notifier_options = []): bool;
 
   /**
-   * Sets arguments to the message entity, overriding existing ones.
-   *
-   * If a message entity has been already set, before calling this method, and
-   * the message entity has arguments, those arguments will be overwritten by
-   * the ones provided to this method. Also, if these arguments are changing an
-   * existing, saved entity, the message entity will be resaved.
+   * Sends a Message based on the given message template to the given user.
    *
+   * @param string $message_template
+   *   The message template ID.
    * @param array $arguments
-   *   The arguments to be passed to the message entity.
+   *   The arguments to be set on the message entity.
+   * @param \Drupal\user\UserInterface $account
+   *   A user account that will be the recipient of the message.
+   * @param array $notifier_options
+   *   An optional associative array of options to pass to the notifier plugin
+   *   that is used (either Email or Digest).
+   * @param bool $digest
+   *   Whether or not to include the message in the user's periodic notification
+   *   digest. If set to TRUE the message will be sent according to the user's
+   *   chosen notification frequency: immediately, daily, weekly or monthly. If
+   *   set to FALSE the message will be sent immediately. Defaults to TRUE.
    *
-   * @return $this
+   * @return bool
+   *   Whether or not the message was sent successfully.
    */
-  public function setArguments(array $arguments): self;
+  public function sendMessageTemplateToUser(string $message_template, array $arguments, UserInterface $account, array $notifier_options = [], bool $digest = TRUE): bool;
```

Comment:
I know it is technically possible to send multiple messages to a single user and keep the digest possibility by looping over the messages in this method, but let's keep it separate to make it easier to understand:

* Need digest? Digests are per user so let's send the messages one by one.
* No digest needed? Feel free to use any of the other methods for mass sending.